### PR TITLE
Introduce Avro plugin

### PR DESCRIPTION
**Pull request by valfirst**

### CODE REVIEW

File: vividus-plugin-avro/src/main/java/org/vividus/avro/steps/AvroSteps.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        try (DataFileReader<Object> dataFileReader = new DataFileReader<>(new SeekableByteArrayInput(data),
+                datumReader))
+        {
+            LOGGER.atInfo()
+                    .addArgument(() -> dataFileReader.getSchema().toString(false))
+                    .log("Avro schema: {}");
+
+            StringBuilder json = new StringBuilder("[");
+
+            Object dataRecord = null;
```
Comment (uarlouski) [inline vividus-plugin-avro/src/main/java/org/vividus/avro/steps/AvroSteps.java:75]: why is this variable needed?

Comment (valfirst) [reply vividus-plugin-avro/src/main/java/org/vividus/avro/steps/AvroSteps.java:75]: see line 78, it's used as a pointer in file

File: vividus-plugin-avro/src/main/java/org/vividus/avro/steps/AvroSteps.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            throws IOException
+    {
+        byte[] data = ResourceUtils.loadResourceOrFileAsByteArray(resourceNameOrFilePath);
+        DatumReader<Object> datumReader = new GenericDatumReader<>();
+        try (DataFileReader<Object> dataFileReader = new DataFileReader<>(new SeekableByteArrayInput(data),
+                datumReader))
+        {
+            LOGGER.atInfo()
+                    .addArgument(() -> dataFileReader.getSchema().toString(false))
+                    .log("Avro schema: {}");
```
Comment (ikalinin1) [inline vividus-plugin-avro/src/main/java/org/vividus/avro/steps/AvroSteps.java:71]:
```suggestion
            LOGGER.atInfo()
                  .addArgument(() -> dataFileReader.getSchema().toString(false))
                  .log("Avro schema: {}");
```

File: vividus-plugin-avro/src/test/java/org/vividus/avro/steps/AvroStepsTests.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    private final TestLogger logger = TestLoggerFactory.getTestLogger(AvroSteps.class);
+
+    @Test
+    void shouldConvertAvroDataToJson() throws IOException
+    {
+        Set<VariableScope> scopes = Set.of(VariableScope.STORY);
+        String variableName = "varName";
+        avroSteps.convertAvroDataToJson("/event-message.avro", scopes, variableName);
+        verify(bddVariableContext).putVariable(scopes, variableName, "[{\"SequenceNumber\": 0, \"Offset\": \"0\", "
```
Comment (ikalinin1) [inline vividus-plugin-avro/src/test/java/org/vividus/avro/steps/AvroStepsTests.java:54]:
```suggestion
        var scopes = Set.of(VariableScope.STORY);
        var variableName = "varName";
        avroSteps.convertAvroDataToJson("/event-message.avro", scopes, variableName);
        verify(bddVariableContext).putVariable(scopes, variableName, "[{\"SequenceNumber\": 0, \"Offset\": \"0\", "
```

File: vividus/src/main/java/org/vividus/bdd/steps/FileSteps.java [hunk 4]
```diff
@@ -53,11 +54,12 @@
      */
     @When("I create temporary file with name `$name` and content `$content` and put path to $scopes variable"
             + " `$variableName`")
-    public void saveResponseBodyToFile(String name, String content, Set<VariableScope> scopes, String variableName)
+    public void createTemporaryFile(String name, DataWrapper content, Set<VariableScope> scopes, String variableName)
```
Comment (ikalinin1) [inline vividus/src/main/java/org/vividus/bdd/steps/FileSteps.java:57]: could you please add an integration test for the case with binary data

Comment (valfirst) [reply vividus/src/main/java/org/vividus/bdd/steps/FileSteps.java:57]: it's added as a part of Avro test

**Review by ikalinin1 — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**